pkg/daemon/file_writers.go

```diff
@@ -45,12 +46,62 @@ func noOrigFileStampName(fpath string) string {
 	return filepath.Join(noOrigParentDir(), fpath+".mcdnoorig")
 }
 
+func isFileOwnedByRPMPkg(fpath string) (string, bool, error) {
+	// Check whether the underlying OS is Linux, skip the rest and return if not (e.g. Mac -> no rpm)
+	// If we cannot find the rpm binary, return an error.
+	path, err := exec.LookPath("rpm")
+	if err != nil {
+		return "", false, err
+	}
+
+	// Check whether the given path is owned by an rpm pkg
+	cmd := exec.Command(path, "-qf", fpath)
+	out, err := cmd.CombinedOutput()
+
+	if err == nil && cmd.ProcessState.ExitCode() == 0 {
+		return strings.TrimSpace(string(out)), true, nil
+	}
+	fileNotOwnedMsg := fmt.Sprintf("file %s is not owned by any package", fpath)
+	if strings.Contains(string(out), fileNotOwnedMsg) {
+		return "", false, nil
+	}
+	return "", false, fmt.Errorf("command %q terminated unexpectedly: %s: %w", cmd, string(out), err)
```

Comment:
nit: instead of terminated, I think you can just say something like `command %q returned with unexpected error:`